pkgs/games/infra-arcana/default.nix

```diff
@@ -0,0 +1,71 @@
+{ stdenv, fetchFromGitLab, cmake, makeWrapper, makeDesktopItem, graphicsmagick,
+  SDL2, SDL2_image, SDL2_mixer }:
+
+stdenv.mkDerivation rec {
+  name = "infra-arcana-${version}";
+  version = "19.2";
+
+  src = fetchFromGitLab {
+    owner = "martin-tornqvist";
+    repo = "ia";
+    rev = "v${version}";
+    sha256 = "0qgfb4djsmh4dbgdz1ahyhkxm6zzhn7vc9xwbvyyb8dzl8abmig7";
+  };
+
+  # The game shows the compile date on the main menu screen. Nix pretends it's
+  # Jan 1 1970 and so it looks funny. This patch hard-codes the datestring to
+  # the release date (taken from git) to preserve determinism while showing
+  # something useful.
+  patches = [ ./release-date.patch ];
+
+  nativeBuildInputs = [ cmake makeWrapper graphicsmagick ];
+
+  buildInputs = [ SDL2 SDL2_image SDL2_mixer ];
+
+  hardeningDisable = [ "fortify" ];
+
+  desktopItem = makeDesktopItem {
+    name = "infra-arcana";
+    desktopName = "Infra Arcana";
+    genericName = "Infra Arcana";
+    comment = meta.description;
+    icon = "infra-arcana";
+    exec = "infra-arcana";
+    categories = "Game;AdventureGame;";
+    extraEntries = ''
+      StartupWMClass=.infra-arcana-wrapped
+    '';
+  };
+
+  postInstall = ''
+    cp -r target/ia/res $out
+    install -D -m 0555 target/ia/ia $out/bin/infra-arcana
+    wrapProgram $out/bin/infra-arcana --run "cd $out"
+
+    i=0; for s in 16 32 48 64 128 256; do
+      iconPath=$out/share/icons/hicolor/$s"x"$s/apps
+      mkdir -p $iconPath
+      gm convert -transparent white ../icon/icon.ico\[$i\] \
+        $iconPath/infra-arcana.png
+      i=$((i+1))
+    done
+
```

Comment:
it looks like this package need missing dependency hicolor-icon-theme .

Reply:
As discussed on IRC I'll submit a patch to include PNG icons to the upstream project. I'll update this PR once done.